bigtable/client/client_options.h

```diff
@@ -81,15 +82,56 @@ class ClientOptions {
   }
 
   /**
-   * Set the grpclb fallback timeout (in ms) for the channel.
+   * Set the grpclb fallback timeout with the timestamp [@p fallback_timeout)
+   * for the channel.
+   *
+   * This function accepts any instantiation of 'std::chrono::duration<>' for
+   * @p duration parameter convert it into milliseconds and pass it to
+   * channel_arguments. For example:
+   *
+   * @code
+   * using namespace std::chrono_literals; // C++14
+   * bigtable::ClientOptions::SetGrpclbFallbackTimeout(std::chrono::milliseconds(5000))
+   * bigtable::ClientOptions::SetGrpclbFallbackTimeout(std::chrono::seconds(5))
+   * @endcode
+   *
+   * The fallback_timeout must not be empty and it should be within the range
+   * of int32. The code will throw exception std::out_of_range if range goes
+   * outside of int32.
+   *
+   * @tparam Rep a placeholder to match the Rep tparam for @p fallback_timeout,
+   *     the semantics of this template parameter are documented in
+   *     `std::chrono::duration<>` (in brief, the underlying arithmetic type
+   *     used to store the number of ticks), for our purposes it is simply a
+   *     formal parameter.
+   * @tparam Period a placeholder to match the Period tparam for @p
+   * fallback_timeout
+   *     , the semantics of this template parameter are documented in
+   *     `std::chrono::duration<>` (in brief, the length of the tick in seconds,
+   *     expressed as a `std::ratio<>`), for our purposes it is simply a formal
+   *     parameter.
+   *
+   * @see
+   * [std::chrono::duration<>](http://en.cppreference.com/w/cpp/chrono/duration)
+   *     for more details.
    *
    * Please see the docs for grpc::ChannelArguments::SetGrpclbFallbackTimeout()
    * on https://grpc.io/grpc/cpp/classgrpc_1_1_channel_arguments.html
    * for more details.
    *
    */
-  void SetGrpclbFallbackTimeout(int fallback_timeout) {
-    channel_arguments_.SetGrpclbFallbackTimeout(fallback_timeout);
+  template <typename Rep, typename Period>
+  void SetGrpclbFallbackTimeout(
+      std::chrono::duration<Rep, Period> fallback_timeout) {
+    std::chrono::milliseconds ft_ms =
+        std::chrono::duration_cast<std::chrono::milliseconds>(fallback_timeout);
+
+    if (ft_ms.count() > std::numeric_limits<int32_t>::max())
+      absl::base_internal::ThrowStdOutOfRange("Duration Exceeds Range for int");
```

Comment:
We should not be using `absl::*internal`.  Please use a plain `throw std::out_of_range` here.  We will fix all of those in a future PR and introduce our own `bigtable::internal::Throw*` functions (tedious, but needs to be done).